src/lightning_app/core/api.py

```diff
@@ -234,6 +236,29 @@ async def post_state(
     api_app_delta_queue.put(DeltaRequest(delta=Delta(deep_diff)))
 
 
+@fastapi_service.put("/api/v1/upload_file/{filename}")
+async def upload_file(filename: str, uploaded_file: UploadFile = File(...)):
+    with TemporaryDirectory() as tmp:
+        drive = Drive(
+            "lit://uploaded_files",
+            component_name="file_server",
+            allow_duplicates=True,
+            root_folder=tmp,
+        )
+        tmp_file = os.path.join(tmp, filename)
+
+        with open(tmp_file, "wb") as f:
+            done = False
+            while not done:
+                # Note: The 8192 number doesn't have a strong reason.
+                content = await uploaded_file.read(8192)
+                f.write(content)
+                done = content == b""
+
+        drive.put(filename)
```

Comment:
should we do `drive.put` in a different process? This might block server thread and since we are using only a single thread for the whole app, it might have effect on healthz, app server end point etc

Reply:
I think it is fine for now. We can always optimize it if needed. I tried with the Training Studio App and it didn't seem to affect the server as shown in the video above.